src/proteus/EvolvingProteus.sol

```diff
@@ -253,16 +259,19 @@ contract EvolvingProteus is ILiquidityPoolImplementation {
       @param px_init The initial price at the x axis
       @param py_final The final price at the y axis
       @param px_final The final price at the y axis
-      @param duration duration for which the curve will evolve
+      @param curveEvolutionStartTime curve evolution start time
+      @param curveEvolutionDuration duration for which the curve will evolve
     */
     constructor(
         int128 py_init,
         int128 px_init,
         int128 py_final,
         int128 px_final,
-        uint256 duration
+        uint256 curveEvolutionStartTime,
+        uint256 curveEvolutionDuration
     ) { 
-        if (duration == 0) revert();
+        if (curveEvolutionDuration == 0) revert();
+        if (curveEvolutionStartTime >= block.timestamp + curveEvolutionDuration) revert();
```

Comment:
i dont understand the reason for this check, why does the start time have to be at least the evolution duration away from deployment time